packages/ocular/src/api/routes/admin/apps/updateApp.ts

```diff
@@ -1,45 +1,76 @@
-import { IsNotEmpty, IsString, IsOptional, IsEnum } from "class-validator";
+import {
+  IsNotEmpty,
+  IsString,
+  IsOptional,
+  IsEnum,
+  ValidateNested,
+} from "class-validator";
+import { Type } from "class-transformer";
 import { validator } from "@ocular/utils";
 import { OAuthService, OrganisationService } from "../../../../services";
 import { AppNameDefinitions } from "@ocular/types";
-import { EventBusService } from "../../../../services";
 const { v4: uuidv4 } = require("uuid");
 
 export default async (req, res) => {
   const validated = await validator(PostAppsReq, req.body);
-  const eventBus_: EventBusService = req.scope.resolve("eventBusService");
   const loggedInUser = req.scope.resolve("loggedInUser");
   const organisationService: OrganisationService = req.scope.resolve(
     "organisationService"
   );
 
-  const link_id = uuidv4();
-
-  const data = {
-    link: validated.link,
-    link_id,
-    status: "processing",
-    emit_event: true,
-  };
+  let data = {};
+  console.log("ORG_ID", loggedInUser);
+  switch (validated.name) {
+    case AppNameDefinitions.WEBCONNECTOR:
+      data = {
+        ...(validated.metadata || {}),
+        emit_event: true,
+        status: "processing",
+        link_id: uuidv4(),
+        org_id: loggedInUser.organisation_id,
+      };
+      break;
 
+    default:
+      data = {
+        ...(validated.metadata || {}),
+      };
+      break;
+  }
+  console.log("API REQUEST DATA", data);
   const installed_apps = await organisationService.updateInstalledApp(
     validated.name,
     data
   );
 
-  if (installed_apps) {
+  if (true) {
```

Comment:
Could you explain what happening here?

Reply:
fixed it , did that for debugging !! resolved now !